src/fsharp/FSharp.Core/async.fs

```diff
@@ -355,7 +343,7 @@ namespace Microsoft.FSharp.Control
 
         let inline fake () = Unchecked.defaultof<AsyncReturn>
 
-        let unfake (_: AsyncReturn)  = ()
+        let inline unfake (_: AsyncReturn)  = ()
```

Comment:
Since we got rid of the other `unfake` function and replaced its uses with `ignore`, couldn't we do the same here? As well as the `fake` function above?

Reply:
I kept `unfake` in `AsyncPrimitives` for clarity; with an `ignore` someone would think that important return values are ignored.

Writing `Unchecked.defaultof<AsyncReturn>` is cumbersome and a `null` isn't helpful.

Reply:
I even tried this:

``` fsharp
[<NoEquality; NoComparison; CompilationRepresentation(CompilationRepresentationFlags.UseNullAsTrueValue)>]
type AsyncReturn =
    | AsyncReturn
    | AsyncReturnDummy of int
```

but the compiler demanded that the internals of `AsyncReturn` be exposed to the signature file.

Reply:
That sounds fine. I still think ignore would be better but it's ok.